submissions/alstep07/friends-app/script.js

```diff
@@ -0,0 +1,159 @@
+		filters.name = null;
+		filters.date = null;
+		filterCards();
+	});
+
+	filterDate.addEventListener("click", ({ target }) => {
+		filters.date = target.id;
+		filters.age = null;
+		filters.name = null;
+		filterCards();
+	});
+
+	resetBtn.addEventListener("click", () => {
+		filters.search = null;
+		filters.gender = null;
+		filters.name = null;
+		filters.age = null;
+		render(results);
+	});
+};
+
+function filterCards() {
+	let filteredArr = data;
+	if (filters.search) filteredArr = sortBySearch(filteredArr, filters.search);
+	if (filters.gender) filteredArr = sortByGender(filteredArr, filters.gender);
+	if (filters.name) filteredArr = sortByName(filteredArr, filters.name);
+	if (filters.age) filteredArr = sortByAge(filteredArr, filters.age);
+	if (filters.date) filteredArr = sortByDate(filteredArr, filters.date);
+	render(filteredArr);
+}
+
+function sortBySearch(arr, str) {
+	return arr.filter((elem) =>
+		`${elem.name.first}${elem.name.last}${elem.location.country}${elem.location.city}`
+			.toLowerCase()
+			.includes(str.toLowerCase())
+	);
+}
+
+function sortByGender(arr, type) {
+	if (type === "all") {
+		return arr;
+	}
+	return arr.filter((elem) => elem.gender === type);
+}
+
+function sortByName(arr, type) {
+	if (type === "name_a-z") {
+		return arr.sort((a, b) => a.name.first.localeCompare(b.name.first));
+	} else if (type === "name_z-a")
+		return arr.sort((a, b) => b.name.first.localeCompare(a.name.first));
+}
+
+function sortByAge(arr, type) {
+	if (type === "age_up") {
+		return arr.sort((a, b) => a.dob.age - b.dob.age);
+	} else if (type === "age_down") {
+		return arr.sort((a, b) => b.dob.age - a.dob.age);
+	}
+}
+
+function sortByDate(arr, type) {
+	if (type === "date_up") {
+		return arr.sort((a, b) => a.registered.age - b.registered.age);
+	} else if (type === "date_down") {
+		return arr.sort((a, b) => b.registered.age - a.registered.age);
+	}
+}
+
+function render(arr) {
+	main.innerHTML = "";
+	const container = document.createDocumentFragment();
+	arr.forEach((elem) => container.append(createCard(elem)));
+	main.append(container);
+}
+
+function createCard(friend) {
+	const card = document.createElement("div");
+	card.innerHTML = `
+		<div class="card ${friend.gender}">
+			<img src="${friend.picture.large}" alt="photo" class="card__img">
+			<p class="card__name">${friend.name.first} ${friend.name.last}</p>
+			<p class="card__age">${friend.dob.age} years</p>
+			<p class="card__location">${friend.location.country}<br>${friend.location.city}</p>
+			<div class="card__contacts">
+				<a class="card__link" href="tel: ${friend.cell}">
+					<img class="card__phone__img" src="${phoneImg}" alt = "phone to">
+				</a>
+				<a class="card__link" href="mailto: ${friend.email}">
+					<img class="card__mail__img" src="${mailImg}" alt = "mail to">
+				</a>
+			</div>
+		</div>`;
+	return card;
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+	fetchFriends(url).then((results) => init(results));
```

Comment:
You can write shorter: 
```javascript
fetchFriends(url).then(init);
```

But your solution is totally fine, this is just a suggestion.

Reply:
May be solution is good, but I can do it better, thank you!